### ISSUE TITLE
Planning release sos-4.7.0

### ISSUE THREAD

**Issue by pmoravec** [sosreport/sos#3529]
Following the commitment to release a new version every 6 months, we shall do the release 4.7.0 around this time. Until there are some objections, I will do the release (hopefully in the right way now :) by following [our checklist](https://github.com/sosreport/sos/wiki/Maintainers-checklist-for-releassing-a-new-version-tag)) during Monday.

Any volunteer to do the release is welcomed, shift in days is no probem, I just want to have the release done in a reasonable time and dont want to step on each other toes.

**Comment by arif-ali** [comment]
sounds good, based on the milestones we were expecting that Monday :)

**Comment by dnegreira** [comment]
Thanks for this @pmoravec! We will be doing our own testing on Ubuntu as soon as the release is done, and after we vetted that we are good to go,  we will start doing the [SRU](https://wiki.ubuntu.com/StableReleaseUpdates) process [for sos](https://wiki.ubuntu.com/SosreportUpdates).

As for the release, even though I am not a maintainer, I would like to at least help out with the one after this one, as this release I don't really have any cycles, but will follow closely. I can definitely make room/set expectations for the next release in August.

**Comment by TurboTurtle** [comment]
Was planning on tackling this, but if you'd like to take point that's fine by me.

**Comment by pmoravec** [comment]
> Was planning on tackling this, but if you'd like to take point that's fine by me.

Well, I do prefer you prepare the release as you already did it perfectly many times, contrary to myself :). Whatever works for you.

**Comment by jcastill** [comment]
@pmoravec I'd like to help as well with anything that's needed. I'm in the same situation as David, I think - not a maintainer here, but maintain sos in RH. So it will be very useful for me to learn and help :)

**Comment by arif-ali** [comment]
An overall collaboration would be great, so the regular folks can also help out. Agreed with David, I was half thinking forward to 4.8.0 in Auguest and would love to be actively involved in preparing that release

**Comment by pmoravec** [comment]
Gladly releasing is relatively simple (until one forgets some step..), the most "difficult" or time consuming is to go through changes since previous release (just the minor one), identify important changes to compose "release notes".

One optional point to the process: it would be nice to announce coming-soon release and ask if there is some pending PR that somebody wants to get in - do it e.g. a week before the day D. I am updating the wiki accordingly.

So.. is there some last merge request before the release? :)

**Comment by arif-ali** [comment]
> Gladly releasing is relatively simple (until one forgets some step..), the most "difficult" or time consuming is to go through changes since previous release (just the minor one), identify important changes to compose "release notes".

Maybe a good idea to have a running release notes somewhere, where a gradual set of items can be added. This way the person doing it, wouldn't need to as much on release day?

> One optional point to the process: it would be nice to announce coming-soon release and ask if there is some pending PR that somebody wants to get in - do it e.g. a week before the day D. I am updating the wiki accordingly.

Yup, good idea :)

**Comment by jcastill** [comment]
You make it sound like it's a tedious process. Isn't there a way to automatize all these tasks in github, perhaps via github actions? or at least part of them?

**Comment by TurboTurtle** [comment]
> > Was planning on tackling this, but if you'd like to take point that's fine by me.
> 
> Well, I do prefer you prepare the release as you already did it perfectly many times, contrary to myself :). Whatever works for you.

I think we both assumed the other was on it ;)

I'm working up the changelog and the commit now. One question for @arif-ali or @dnegreira  though - do you need us to also bump the `debian/changelog` file like we do the spec file? Would it look like this?

```
sosreport (4.7.0-0ubuntu1) mantic; urgency=medium

* New 4.7.0 upstram release

-- Jake Hunsaker <jacob.r.hunsaker@gmail.com> Mon, 19 Feb 2024 18:15:00 +0000

sosreport (4.6.0-0ubuntu1) mantic; urgency=medium

  * New 4.6.0 upstream.

 -- Nikhil Kshirsagar <nikhil.kshirsagar@canonical.com>  Thu, 17 Aug 2023 08:17:20 +0000
```

Or do you prefer to handle this file in your downstream pipeline?

**Comment by arif-ali** [comment]
This was discussed in #3492, and gave an example on how we can change it.

It would be good to do it here, as the package built in CI will be consistent.

We will have a separate changelog in our pipeline, as that does work differently and we have to add a few extra items per distro version.

Hope that helps

**Comment by pmoravec** [comment]
I added the step to the Wiki https://github.com/sosreport/sos/wiki/Maintainers-checklist-for-releasing-a-new-version-or-tag .

Thanks @TurboTurtle for preparing the release, I was unable to participate here due to several independent reasons since Monday until now.

I plan to release `sos-4.7.1` (by myself) in approx one month - I should notify a week ahead, until I forget.

Closing this as resolved, thanks all for collaboration.
